connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/KeyValueGroupedDataset.scala

```diff
@@ -879,7 +879,7 @@ class KeyValueGroupedDataset[K, V] private[sql] () extends Serializable {
    *
    * See [[Encoder]] for more details on what types are encodable to Spark SQL.
    */
-  def transformWithState[U: Encoder, S: Encoder](
+  private[sql] def transformWithState[U: Encoder, S: Encoder](
```

Comment:
why do we make this API private?

Reply:
I found these methods are private in Spark Core: https://github.com/apache/spark/blob/f99291ac864d8902aa131fa93d46725a47d88263/sql/core/src/main/scala/org/apache/spark/sql/KeyValueGroupedDataset.scala#L663.